easybuild/toolchains/iccifortcuda.py

```diff
@@ -38,3 +39,22 @@ class IccIfortCUDA(IccIfort, Cuda):
 
     COMPILER_MODULE_NAME = IccIfort.COMPILER_MODULE_NAME + Cuda.COMPILER_CUDA_MODULE_NAME
     SUBTOOLCHAIN = IccIfort.NAME
+
+    def is_dep_in_toolchain_module(self, name):
+        """Check whether a specific software name is listed as a dependency in the module for this toolchain."""
+        # icc & ifort do not need to be actual dependencies in iccifort module,
+        # since they could also be installed together in a single directory.
+        # Let IccIfort check that.
+        res = IccIfort.is_dep_in_toolchain_module(self, name)
+
+        # Also check for CUDA since this is IccIfortCUDA toolchain
+        # as long as the corresponding $EBROOT* and $EBVERSION* environment variables are defined, it should be OK
+        if not res:
+            if name in 'CUDA':
+                self.log.info("Checking whether %s is a toolchain component even though it is not a dependency", name)
```

Comment:
This is not correct, since `CUDA` *is* a direct dep of `iccifortcuda`?

This makes me wonder why this custom `is_dep_in_toolchain_module` is actually needed at all for `iccifortcuda`...
The `super` call in `IccIfort.is_dep_in_toolchain_module` is supposed to take care of this case?

Reply:
The log.info text is also just a c&p from iccifort.py. The tests failed when I did not have this code in there. Not sure why.

Reply:
Ah yes.... probably because how i wrote the iccifortcuda test...

Reply:
The log message is correct in `iccifort.py` (since there `icc` and `ifort` may indeed not be actual deps), but it's wrong here for `CUDA`, so just remove that log line?

So, is this custom `is_dep_in_toolchain_module` only required to pass the new test, but not outside? If so, we should fix the test instead... :)

Reply:
It's probably needed for the same basic reason that you need it in iccifort.py, but it should probably not need the "if name == 'CUDA'" part.
But I don't know how to rewrite the test for that. I didn't quite understand how the test worked, i.e. what exactly  happens at each step of the test?